test/NuGet.Core.Tests/NuGet.Configuration.Test/PackageSourceProviderTests.cs

```diff
@@ -2271,6 +2271,34 @@ public void PackageSourcesChanged_EventRunsSubscriptions(bool subscribeToEvent)
             Assert.Equal(subscribeToEvent, eventRun);
         }
 
+        [Fact]
+        public void GetPackageSourceBySource_TwoSourcesWithSameUrl_ReturnsFirstSource()
+        {
+            // Arrange
+            using TestDirectory testDirectory = TestDirectory.Create();
+
+            const string sourceUrl = "https://contoso.test/nuget/index.json";
+            const string contents = $@"<configuration>
```

Comment:
I think we can leverage the `CreateNuGetConfigContent` method to create NuGet.Config file content by simply passing the package sources. In my opinion, the said method has room for improvement, which is beyond the scope of this PR. For instance, an improvement could be to create the nuget.config string just once (cache it), and then invoke the `Replace` method as needed. This is effective because the `Replace` method creates a new string anyway.

Reply:
There's also a `SettingsTestUtils.CreateConfigurationFile`